iceoryx_posh/include/iceoryx_posh/gateway/gateway_generic.hpp

```diff
@@ -149,6 +149,8 @@ class GatewayGeneric : public gateway_t
     std::thread m_discoveryThread;
     std::thread m_forwardingThread;
 
+    capro::Interfaces m_interface = capro::Interfaces::INTERFACE_END;
```

Comment:
Please don't do this. This enum value is used for sanity checks on deseralization. If you have an optional interface please use a `cxx::optional`

Reply:
This value is not optional and must be set via the constructor which already happens. Lets assume someone writes a new constructor and forgets to set this variable then it is uninitialized and we have undefined behavior which is worse than to have this defined value. 

This is the intention of this line to extinguish undefined behavior in the case new constructors are added.

Reply:
@elfenpiff The best alternative I can currently think of is to shift the responsibility to someone else. Since this is already passed to the base class `gateway_t`, why not require that class to have a `getInterface` method?

-  `gateway_t` is a `GatewayBase`
-  `GatewayBase´ has an `m_interfaceImpl`
-  `m_interfaceImpl` can be use to get the `Interface` by `m_interfaceImpl.getCaProServiceDescription().getSourceInterface()`

What do you think of this?

Reply:
@elBoberido yeah this makes more sense, I updated it.